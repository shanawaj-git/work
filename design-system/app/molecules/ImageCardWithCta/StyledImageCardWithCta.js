import styled from 'styled-components';
import { colorSelector } from 'themes';

export const ImageCardWithCtaWrapper = styled.div`
  display: flex;
  background: ${colorSelector('c200')};
  box-shadow: 0px 3px 20px rgba(96, 108, 116, 0.13);
  border-radius: 10px;
  padding: 24px 16px;

  .image-card-title,
  .image-card-subTitle {
    color: ${colorSelector('fontPrimary')};
  }

  .image-card-icon {
    width: 24px;
    height: 24px;
  }

  .image-card-content {
    margin-left: 13px;
  }

  .image-card-button {
    background: none;
    margin-left: auto;
    width: auto;
    height: auto;
    img {
      width: 24px;
      height: 24px;
    }
  }
`;
