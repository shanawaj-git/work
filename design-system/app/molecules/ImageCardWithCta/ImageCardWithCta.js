import React from 'react';
import PropTypes from 'prop-types';
import Typography, { TypographyType } from 'atoms/Typography';
import { ImageCardWithCtaWrapper } from './StyledImageCardWithCta';

const ImageCardWithCta = props => {
  const { className, cardIcon, title, subTitle, buttonIcon, onClick } = props;
  return (
    <div className={className}>
      <ImageCardWithCtaWrapper>
        <img className="image-card-icon" src={cardIcon} alt={title} />

        <div className="image-card-content">
          <Typography
            className="image-card-title"
            typographyType={TypographyType.HEAD_LINE}
          >
            {title}
          </Typography>
          <Typography
            className="image-card-subTitle"
            typographyType={TypographyType.CAPTION}
          >
            {subTitle}
          </Typography>
        </div>
        {buttonIcon && (
          <button type="button" className="image-card-button" onClick={onClick}>
            <img src={buttonIcon} alt={title} />
          </button>
        )}
      </ImageCardWithCtaWrapper>
    </div>
  );
};
export default ImageCardWithCta;

ImageCardWithCta.propTypes = {
  className: PropTypes.string,
  cardIcon: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  buttonIcon: PropTypes.string,
  onClick: PropTypes.func,
};
