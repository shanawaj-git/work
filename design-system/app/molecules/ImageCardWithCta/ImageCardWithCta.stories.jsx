import React from 'react';
import ImageCardWithCta from './ImageCardWithCta';
import StoryWrapper from 'molecules/utility/StoryWrapper';

export default {
  title: 'Components/Cards/ImageCardWithCta',
  component: ImageCardWithCta,
  argTypes: {
    themeType: {
      control: { type: 'select', options: ['DEFAULT', 'MPC'] },
    },
    className: { table: { disable: true } },
    onClick: { table: { disable: true } },
  },
};

const Template = args => (
  <StoryWrapper themeVariant={args.themeType}>
    <ImageCardWithCta {...args} />
  </StoryWrapper>
);

export const PrimaryImageCardWithCta = Template.bind({});

const icon = require('images/location-icon.svg');
const buttonIcon = require('images/arow-right.svg');

PrimaryImageCardWithCta.args = {
  cardIcon: icon,
  title: 'We Work offices',
  subTitle: 'The Offices 4, Albatha Next, 8th floor...',
  buttonIcon: buttonIcon,
};
