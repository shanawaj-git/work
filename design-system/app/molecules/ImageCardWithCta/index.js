import ImageCardWithCta from './ImageCardWithCta';

export default ImageCardWithCta;
