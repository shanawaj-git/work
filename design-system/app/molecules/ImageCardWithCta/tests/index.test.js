import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { render, unmountComponentAtNode } from 'react-dom';
import renderer, { act } from 'react-test-renderer';
import ImageCardWithCta from '../index';

const shallowRenderer = new ShallowRenderer();

let container = null;

const icon = require('images/location-icon.svg');
const buttonIcon = require('images/arow-right.svg');

const defualtProps = {
  icon,
  title: 'We Work offices',
  subTitle: 'The Offices 4, Albatha Next, 8th floor...',
  buttonIcon,
  onClick: jest.fn(),
};

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('ImageCardWithCta.js', () => {
  describe('index', () => {
    it('Should import succesfully', () => {
      const tree = renderer.create(<ImageCardWithCta {...defualtProps} />);
      expect(tree).toBeTruthy();
    });

    it('Should display ImageCardWithCta item with icon, title, subTitle and button with icon', () => {
      shallowRenderer.render(<ImageCardWithCta {...defualtProps} />);
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });

    it('Should be able to click on the arrow button in the card ', () => {
      act(() => {
        render(<ImageCardWithCta {...defualtProps} />, container);
      });

      const cardButton = document.querySelector('.image-card-button');

      act(() => {
        cardButton.dispatchEvent(new MouseEvent('click', { bubbles: true }));
      });

      expect(defualtProps.onClick).toHaveBeenCalledTimes(1);
    });
  });
});
