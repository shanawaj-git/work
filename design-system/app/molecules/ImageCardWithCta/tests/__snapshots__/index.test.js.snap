// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCardWithCta.js index Should display ImageCardWithCta item with icon, title, subTitle and button with icon 1`] = `
<div>
  <ForwardRef(StyledImageCardWithCta__ImageCardWithCtaWrapper)>
    <img
      alt="We Work offices"
      className="image-card-icon"
    />
    <div
      className="image-card-content"
    >
      <Typography
        className="image-card-title"
        typographyType="headLine"
      >
        We Work offices
      </Typography>
      <Typography
        className="image-card-subTitle"
        typographyType="caption"
      >
        The Offices 4, Albatha Next, 8th floor...
      </Typography>
    </div>
    <button
      className="image-card-button"
      onClick={[MockFunction]}
      type="button"
    >
      <img
        alt="We Work offices"
        src="IMAGE_MOCK"
      />
    </button>
  </ForwardRef(StyledImageCardWithCta__ImageCardWithCtaWrapper)>
</div>
`;
